CHANGELOG.md

```diff
@@ -89,6 +89,7 @@
 * [ENHANCEMENT] Update Docker base images from `alpine:3.17.3` to `alpine:3.18.0`. #5065
 * [ENHANCEMENT] Compactor: reduced the number of "object exists" API calls issued by the compactor to the object storage when syncing block's `meta.json` files. #5063
 * [ENHANCEMENT] Distributor: Push request rate limits (`-distributor.request-rate-limit` and `-distributor.request-burst-size`) and their associated YAML configuration are now stable. #5124
+* [ENHANCEMENT] Go: updated to 1.20.5. #5185
```

Comment:
Does this need to go in the changelog? It's not user visible.

Reply:
The update to 1.20.4 got a [changelog entry](https://github.com/grafana/mimir/blob/main/CHANGELOG.md?plain=1#L66), hence I mirrored that on this PR. I also get confused by when we need one and when we don't.

Reply:
Since @pstibrany opened the previous pull request for 1.20.4, and both the .4 and .5 point releases address security issues, I think we should be ok with having the changelog entry here. In doubt, probably better to err on the side of verbosity?

Reply:
I'm in favour of keeping the changelog entry :+1:

Reply:
I see that 1.20.4 is _also_ listed among the 2.9.0-rc.0 changes, I think that entry should be removed (since in effect this version will update to Go 1.20.5).